images/common/bootstrap.sh

```diff
@@ -335,12 +335,85 @@ bootstrap_c8y() {
     echo "Uploading certificate to Cumulocity using tedge"
     tedge cert upload c8y
 
-    # Grace period for the server to process the certificate
-    # but it is not critical for the connection, as the connection
-    # supports automatic retries, but it can improve the first connection success rate
+        # Grace period for the server to process the certificate
+        # but it is not critical for the connection, as the connection
+        # supports automatic retries, but it can improve the first connection success rate
     sleep "$UPLOAD_CERT_WAIT"
 }
 
+bootstrap_c8y_basic() {
+    sudo tedge config set c8y.auth_method basic
+            # Device id
+    DEVICE_ID=$(prompt_value "Enter the device.id" "$DEVICE_ID")
+            # Cumulocity URL
+    C8Y_BASEURL=$(prompt_value "Enter the Cumulocity IoT url" "$C8Y_BASEURL")
+    if [ -n "$C8Y_BASEURL" ]; then
+        C8Y_BASEURL=$(parse_domain "$C8Y_BASEURL")
+    fi
+            # Cumulocity username & password
+    C8Y_USER=$(prompt_value "Enter the Cumulocity username: " "$C8Y_USER")
+    if [ "$SHOULD_PROMPT" = 1 ] && [ "$CAN_PROMPT" = 1 ]; then
+        printf "Enter the Cumulocity password: " >&2
+        stty -echo
+        read -r C8Y_PASSWORD
+        stty echo
+        echo
+    fi
+            # Set device id and url
+    sudo tedge config set c8y.device.id "$DEVICE_ID"
+    sudo tedge config set c8y.url "$C8Y_BASEURL"
+    
+    if [ ! -f /etc/tedge/credentials.toml ]; then
+        sudo touch /etc/tedge/credentials.toml
+        sudo chmod 600 /etc/tedge/credentials.toml
+    fi
```

Comment:
It might make more sense executing `sudo chmod 600 /etc/tedge/credentials.toml` after the credentials are written to file. and then you can get rid of the if statement (as you generally want to be applying the permissions every time you bootstrap).